### PR TITLE
feat: allow user to set and update default gas price

### PR DESCRIPTION
**Pull request by cmwaters**
Closes: #3537

This PR allows the default gas price to be set in the constructor and updated.

This PR also makes a modification to `SetupTxClient` which will query the chain for the current minimum gas price, taking the maximum of the local and global min gas price

### CODE REVIEW

File: pkg/user/tx_client.go [hunk 1]
```diff
@@ -40,6 +45,13 @@ func WithGasMultiplier(multiplier float64) Option {
 	}
 }
 
+// WithGasPrice sets the price the
```
Comment (rootulp) [inline pkg/user/tx_client.go:48]:
```suggestion
// WithGasPrice sets the gas price
```

File: pkg/user/tx_client.go [hunk 2]
```diff
@@ -190,6 +211,8 @@ func (client *TxClient) SubmitPayForBlobsWithAccount(ctx context.Context, accoun
 
 // BroadcastPayForBlob signs and broadcasts a transaction to pay for blobs.
 // It does not confirm that the transaction has been committed on chain.
+// If no gas or gas price is set, it will estimate the gas and use
+// the provided gas price
```
Comment (rootulp) [inline pkg/user/tx_client.go:215]: "the provided" seems confusing here b/c the first part of this sentence says "if no gas or gas price is set". Could it be rephrased to something else?

```suggestion
// the max effective gas price: max(localMinGasPrice, globalMinGasPrice).
```

**Review by rootulp — APPROVED** [review]

File: pkg/user/tx_client.go [hunk 3]
```diff
@@ -511,3 +537,60 @@ func (client *TxClient) getAccountNameFromMsgs(msgs []sdktypes.Msg) (string, err
...
+	client.gasPrice = price
+}
+
+func (client *TxClient) SetGasMultiplier(multiplier float64) {
+	client.mtx.Lock()
+	defer client.mtx.Unlock()
+	client.gasMultiplier = multiplier
+}
+
+// queryMinimumGasPrice queries both the nodes local and network wide
```
Comment (ninabarbakadze) [inline pkg/user/tx_client.go:553]:
```suggestion
// QueryMinimumGasPrice queries both the nodes local and network wide
```

File: pkg/user/tx_client.go [hunk 4]
```diff
@@ -40,6 +45,13 @@ func WithGasMultiplier(multiplier float64) Option {
 	}
 }
 
+// WithGasPrice sets the gas price
```
Comment (ninabarbakadze) [inline pkg/user/tx_client.go:48]:
```suggestion
// WithGasPrice sets the gas price.
```

File: pkg/user/tx_client.go [hunk 5]
```diff
@@ -511,3 +537,60 @@ func (client *TxClient) getAccountNameFromMsgs(msgs []sdktypes.Msg) (string, err
...
+	}
+
+	// return the highest value of the two
+	if globalMinPrice > localMinPrice {
+		return globalMinPrice, nil
+	}
+	return localMinPrice, nil
+}
+
+func QueryGlobalMinGasPrice(ctx context.Context, grpcConn *grpc.ClientConn) (float64, error) {
```
Comment (ninabarbakadze) [inline pkg/user/tx_client.go:579]: godoc missing

**Review by ninabarbakadze — COMMENTED** [review]
do we not need tests for these?

File: pkg/user/tx_client.go (hunk 4 above)
Comment (cmwaters) [reply pkg/user/tx_client.go:48]: lol

**Comment by cmwaters** [comment]
> do we not need tests for these?

The querying is somewhat covered by existing tests. We query both local and global when we initialize the client. 

I could however add tests for the options that manually specify it

**Review by rootulp — APPROVED** [review]

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Review by rootulp — APPROVED** [review]

**Pull request merged**